suite-native/atoms/src/Button/IconButton.tsx

```diff
@@ -1,93 +1,108 @@
-            },
-            medium: {
-                width: 32,
-                height: 32,
-                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
-            },
-            large: {
-                width: 48,
-                height: 48,
-                borderRadius: isRounded ? utils.borders.radii.round : utils.borders.radii.small,
-            },
-        };
         return {
-            justifyContent: 'center',
-            alignItems: 'center',
-            ...colorSchemeStyles[colorScheme],
-            ...sizeStyles[size],
+            padding: 0,
+            height: sizeDimensions[size],
+            width: hasTitle ? 'auto' : sizeDimensions[size],
         };
-    },
-);
-
-const getIconColor = (colorScheme: ButtonColorScheme) => {
-    switch (colorScheme) {
-        case 'gray':
-            return 'gray700';
-        case 'red':
-            return 'red';
-        default:
-            return 'gray0';
-    }
-};
+    }),
+]);
 
 export const IconButton = ({
     iconName,
-    onPress,
     style,
-    colorScheme = 'primary',
+    title,
+    colorSchemeName = 'primary',
     size = 'medium',
-    isRounded = false,
+    isDisabled = false,
+    ...pressableProps
 }: IconButtonProps) => {
+    const [isPressed, setIsPressed] = useState(false);
     const { applyStyle } = useNativeStyles();
+    const { textColor, disabledTextColor, backgroundColor, onPressColor } =
+        buttonSchemeToColorsMap[colorSchemeName];
+
+    const animatedPressStyle = useButtonPressAnimatedStyle(
+        isPressed,
+        backgroundColor,
+        onPressColor,
+    );
+
+    const iconColor = isDisabled ? disabledTextColor : textColor;
 
+    const handlePressIn = () => setIsPressed(true);
+    const handlePressOut = () => setIsPressed(false);
     return (
-        <TouchableOpacity
-            onPress={onPress}
-            style={[applyStyle(iconButtonStyle, { isRounded, size, colorScheme }), style]}
-        >
-            <Icon name={iconName} color={getIconColor(colorScheme)} size={size} />
-        </TouchableOpacity>
+        <Box alignItems="center" style={style}>
+            <Pressable
+                onPressIn={handlePressIn}
+                onPressOut={handlePressOut}
+                disabled={isDisabled}
+                {...pressableProps}
+            >
+                <Animated.View
+                    style={[
+                        animatedPressStyle,
+                        applyStyle(iconButtonStyle, {
+                            size,
+                            colorSchemeName,
+                            isDisabled,
+                            hasTitle: !!title,
+                        }),
+                    ]}
+                >
+                    <Icon name={iconName} color={iconColor} size={size} />
+                </Animated.View>
+            </Pressable>
+            <Pressable
```

Comment:
Do you need two Pressable for this? Why not wrap it in just one?

Reply:
fixed in ecbbd64b5